webapp/src/tutorial.tsx

```diff
@@ -38,7 +43,9 @@ export function getUsedBlocksAsync(code: string[], id: string, language?: string
             .then(entry => {
                 if (entry?.blocks && Object.keys(entry.blocks).length > 0 && !skipCache) {
                     pxt.tickEvent(`tutorial.usedblocks.indexeddb`, { tutorial: id });
-                    return Promise.resolve(entry.blocks);
+                    // populate snippets if usedBlocks are present, but snippets are not
+                    if (!entry?.snippets) getUsedBlocksInternalAsync(code, id, language, db, skipCache);
```

Comment:
should this be returned?

Reply:
nope, it only needs the usedblocks for the tutorial filter. this just populates the snippets so they're cached next time. (snippetblocks aren't used anywhere yet, they'll be pulled into jasmine's stuff later)